sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -766,6 +778,15 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
     }
   }
 
+  /**
+   * A convenient function for paths validation in APIs.
+   */
+  private def assertNonEmptyPaths(paths: String*): Unit = {
+    if (userSpecifiedSchema.isEmpty) {
+      require(paths.nonEmpty, "The paths cannot be empty")
```

Comment:
I am actually still not sure on this. What about `DataFrameReader.load()`? I feel like it's better to delegate this check to DataSources themselves.

Reply:
What about SQL?

Reply:
> I am actually still not sure on this. What about `DataFrameReader.load()`? I feel like it's better to delegate this check to DataSources themselves.

`DataFrameReader.load()`
<img width="1015" alt="image" src="https://github.com/apache/spark/assets/15246973/135a7936-7cce-4c60-8c70-280e4cc7538f">

Reply:
> What about SQL?

About SQL, as follows:
<img width="613" alt="image" src="https://github.com/apache/spark/assets/15246973/8c2ed6fe-16e8-4fbf-913f-716b8ef56ed0">

Reply:
Can you try to `CREATE TABLE ... USING` syntax, and select from it?